lib/order.rb

```diff
@@ -0,0 +1,93 @@
+require_relative 'menu'
+require_relative 'text'
+
+class Order
+
+  attr_reader :items, :total
+   
+  def initialize(menu_class = Menu)
+    @items = []
+    @total = 0
+    @menu = menu_class.new
+  end
+  
+  def add_item(item, quantity = 1)
+    item_in_order?(item) ? increase_quantity(item, quantity) : add_new_item(item, quantity)
+    increase_total(item, quantity)
+  end
+
+  def remove_item(item, quantity = 1)
+    item_in_order?(item) ? quantity = quantity_validator(item, quantity) : quantity = 0
+    decrease_quantity(item, quantity)
+    remove_zero_quantity
+    decrease_total(item, quantity)
+  end
+
+  def check_sum
+    @total == create_bill
+  end
+
+  def place_order(total = @total, items = @items, text_class = Text)
+    check_sum ? text_class.new(items, total) : "The total does not match the bill"
+  end
+
+  private 
+
+  def quantity_validator(item, quantity)
+    @items.each do |item0|
+      if item0[:name] == item
+        quantity = item0[:quantity] if item0[:quantity] < quantity
+      end
+    end
+    quantity
+  end
+
+  def create_bill
+    bill = 0
+    @items.each do |item|
+      @menu.menu_list.each do |dish|
+        bill += dish[:price] * item[:quantity] if dish[:name] == item[:name]
+      end
+    end
+    bill
+  end
+
+  def increase_quantity(item, quantity)
+    @items.each do |dish|
+      dish[:quantity] += quantity if item == dish[:name]
+    end
+  end
+
+  def decrease_quantity(item, quantity)
+    @items.each do |dish|
+      dish[:quantity] -= quantity if item == dish[:name]
+    end
+  end
+
+  def increase_total(item, quantity)
+    @menu.menu_list.each do |dish|
+      @total += (dish[:price] * quantity) if item == dish[:name]
+    end
+  end
+
+  def decrease_total(item, quantity)
```

Comment:
Maybe the decrease_total method should state "Your basket is empty" or another statement rather than take the total amount to negative?